src/jira/client/jira-client-issue-key-helper.ts

```diff
@@ -0,0 +1,172 @@
+import { uniq } from "lodash";
+import Logger from "bunyan";
+import { createHashWithSharedSecret } from "utils/encryption";
+import {
+	JiraAssociation,
+	JiraCommit,
+	JiraRemoteLink
+} from "interfaces/jira";
+
+export interface IssueKeyObject {
+	issueKeys?: string[];
+	associations?: JiraAssociation[];
+}
+
+// Max number of issue keys we can pass to the Jira API
+export const ISSUE_KEY_API_LIMIT = 500;
+
+/**
+ * Truncates to ISSUE_KEY_API_LIMIT elements in an array
+ */
+export const truncate = (array: unknown[]) => {
+	return array.slice(0, ISSUE_KEY_API_LIMIT);
+};
+
+/**
+ * Truncates branches, commits and PRs to their first ISSUE_KEY_API_LIMIT issue keys
+ */
+export const truncateIssueKeys = (repositoryObj) => {
+	updateRepositoryIssueKeys(repositoryObj, truncate);
+};
+
+/**
+ * Get truncated issue keys and associations based on the ISSUE_KEY_API_LIMIT.
+ */
+export const getTruncatedIssueKeys = (data: IssueKeyObject[] = []): IssueKeyObject[] =>
```

Comment:
This function was rewritten, during testing it was found that the existing function wasn't truncating keys